hazelcast/src/test/java/com/hazelcast/config/XmlOnlyConfigBuilderTest.java

```diff
@@ -0,0 +1,198 @@
+        }
+    }
+
+    @Test
+    public void testConfig2Xml2DefaultConfig() {
+        testConfig2Xml2Config("hazelcast-default.xml");
+    }
+
+    @Test
+    public void testConfig2Xml2FullConfig() {
+        testConfig2Xml2Config("hazelcast-fullconfig.xml");
+    }
+
+    @Test
+    public void testXSDDefaultXML() throws Exception {
+        testXSDConfigXML("hazelcast-default.xml");
+    }
+
+    @Test
+    public void testFullConfig() throws Exception {
+        testXSDConfigXML("hazelcast-fullconfig.xml");
+    }
+
+    // No YAML identical to the XML can be written
+    @Test(expected = IllegalArgumentException.class)
+    public void testAttributeConfig_noName_noExtractor() {
+        String xml = HAZELCAST_START_TAG
+                + "   <map name=\"people\">\n"
+                + "       <attributes>\n"
+                + "           <attribute></attribute>\n"
+                + "       </attributes>"
+                + "   </map>"
+                + HAZELCAST_END_TAG;
+        buildConfig(xml);
+    }
+
+    // No YAML identical to the XML can be written
+    @Test(expected = IllegalArgumentException.class)
+    public void testAttributeConfig_noName_noExtractor_singleTag() {
+        String xml = HAZELCAST_START_TAG
+                + "   <map name=\"people\">\n"
+                + "       <attributes>\n"
+                + "           <attribute/>\n"
+                + "       </attributes>"
+                + "   </map>"
+                + HAZELCAST_END_TAG;
+        buildConfig(xml);
+    }
+
+    private void assertXsdVersion(String buildVersion, String expectedXsdVersion) {
+        System.setProperty(HAZELCAST_INTERNAL_OVERRIDE_VERSION, buildVersion);
+        assertEquals("Unexpected release version retrieved for build version " + buildVersion, expectedXsdVersion,
+                new XmlConfigBuilder().getReleaseVersion());
+    }
+
+    private static void testConfig2Xml2Config(String fileName) {
+        String pass = "password";
+        Config config = new ClasspathXmlConfig(fileName);
+        config.getGroupConfig().setPassword(pass);
+
+        String xml = new ConfigXmlGenerator(true, false).generate(config);
+        Config config2 = new InMemoryXmlConfig(xml);
+
+        assertTrue(ConfigCompatibilityChecker.isCompatible(config, config2));
+    }
+
+    private Config buildConfig(String xml) {
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
+        return configBuilder.build();
+    }
+
+    private void testXSDConfigXML(String xmlFileName) throws Exception {
+        SchemaFactory factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        URL schemaResource = XMLConfigBuilderTest.class.getClassLoader().getResource("hazelcast-config-"
+                + Versions.CURRENT_CLUSTER_VERSION + ".xsd");
+        assertNotNull(schemaResource);
```

Comment:
I know it's copied, but in general having assertions in the middle of the test is a code smell. Tests should have always the structure: // given // when // then. Anyway, feel free to keep it, as I know it's just a copy from the other class.

Reply:
Removed the assertion. I actually don't see the reason for asserting on this, since we expect it to be non-null and null anyways breaks the test.